engine/src/main/java/nl/inl/blacklab/search/results/ResultsAbstract.java

```diff
@@ -0,0 +1,294 @@
+package nl.inl.blacklab.search.results;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Random;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+import nl.inl.blacklab.exceptions.BlackLabRuntimeException;
+import nl.inl.blacklab.resultproperty.ResultProperty;
+import nl.inl.blacklab.search.BlackLab;
+import nl.inl.blacklab.search.BlackLabIndex;
+import nl.inl.blacklab.search.indexmetadata.AnnotatedField;
+import nl.inl.util.ThreadAborter;
+
+/**
+ * A list of results of some type.
+ *
+ * All subclasses should be thread-safe.
+ *
+ * @param <T> result type, e.g. Hit
+ */
+public abstract class ResultsAbstract<T, P extends ResultProperty<T>> implements Results<T, P> {
+
+    /** Id the next Hits instance will get */
+    private static int nextHitsObjId = 0;
+
+    private static synchronized int getNextHitsObjId() {
```

Comment:
It'll be much faster to use AtomicInteger's incrementAndGet here.

Reply:
Will do that, thanks!